pkg/rules/rules.go

```diff
@@ -29,7 +30,7 @@ func (r RuleNamespace) LintPromQLExpressions() (int, int, error) {
 	for i, group := range r.Groups {
 		for j, rule := range group.Rules {
 			log.WithFields(log.Fields{"rule": getRuleName(rule)}).Debugf("linting PromQL")
-			exp, err := parser.ParseExpr(rule.Expr)
+			exp, err := promql.ParseExpr(rule.Expr)
```

Comment:
what necessitated this change?

Reply:
I don't have a whole lot of history here, but I think it refers to https://github.com/cortexproject/cortex/pull/1155. It seems have copied code over to avoid upgrading the Prometheus parser/promql package.

I think this has changed a while back and we haven't been ready to upgrade yet.